internal/weapons/catalyst/perception/perception.go

```diff
@@ -75,38 +69,33 @@ func NewWeapon(c *core.Core, char *character.CharWrapper, p weapon.WeaponProfile
 	cd := (13 - r) * 60
 	dmg := 2.1 * float64(r) * 0.3
 
-	c.Events.Subscribe(event.OnAttackWillLand, func(args ...interface{}) bool {
+	w.ai = combat.AttackInfo{
+		ActorIndex: char.Index,
+		Abil:       "Eye of Preception Proc",
+		AttackTag:  combat.AttackTagWeaponSkill,
+		ICDTag:     combat.ICDTagNone,
+		ICDGroup:   combat.ICDGroupDefault,
+		Element:    attributes.Physical,
+		Durability: 100,
+		Mult:       dmg,
+	}
+
+	c.Events.Subscribe(event.OnDamage, func(args ...interface{}) bool {
 		ae := args[1].(*combat.AttackEvent)
 		if ae.Info.ActorIndex != char.Index {
 			return false
 		}
+		if ae.Info.AttackTag != combat.AttackTagNormal && ae.Info.AttackTag != combat.AttackTagExtra {
+			return false
+		}
 		if char.StatusIsActive(icdKey) {
 			return false
 		}
 		char.AddStatus(icdKey, cd, true)
 
-		ai := combat.AttackInfo{
-			ActorIndex: char.Index,
-			Abil:       "Eye of Preception Proc",
-			AttackTag:  combat.AttackTagWeaponSkill,
-			ICDTag:     combat.ICDTagNone,
-			ICDGroup:   combat.ICDGroupDefault,
-			Element:    attributes.Physical,
-			Durability: 100,
-			Mult:       dmg,
-		}
-		w.atk = &combat.AttackEvent{
-			Info:     ai,
-			Snapshot: char.Snapshot(&ai),
```

Comment:
Is this an intentional change to make Eye effectively dynamic instead of snapshotting?